packages/purpose-process/src/routers/PurposeRouter.ts

```diff
@@ -308,7 +308,7 @@ const purposeRouter = (
       const ctx = fromAppContext(req.ctx);
 
       try {
-        validateAuthorization(ctx, [ADMIN_ROLE]);
+        validateAuthorization(ctx, [ADMIN_ROLE, M2M_ADMIN_ROLE]);
 
         const purposeVersion = await purposeService.createPurposeVersion(
```

Comment:
I think this function should return also the metadata and the entire purpose so we follow the standard of polling for the metadata of the resource (the purpose) getting the expected version from the response